### PR TITLE
Surface error occurring during import of @parcel/watcher

### PR DESCRIPTION
**Pull request by spawnia**
## Description

Follow up from https://github.com/dotansimha/graphql-code-generator/pull/9506, which provides the following error message when failing to import `@parcel/watcher`:

```
$ graphql-codegen --project=limes --watch
✔ Parse Configuration
✔ Generate outputs
  Parcel watcher not found. To use this feature, please make sure to provide @parcel/watcher as a peer dependency.
```

I think the provided error message is unclear and misleading in a couple of ways:
1. It suggests to provide `@parcel/watcher` as a peer dependency. It makes sense for it to be a peer dependency of this package, but unless the user itself is distributing a package, they should just install the package.
2. It swallows any underlying error during the import that could prevent `@parcel/watcher` from working, even if installed.

In my environment, simply installing `@parcel/watcher` as a dependency was not sufficient. After some debugging, I modified the piece of code that produces the error message to append the thrown `err` and found that the cause was more detailed:

```
$ graphql-codegen --project=limes --watch
✔ Parse Configuration
✔ Generate outputs
  Parcel watcher not found. To use this feature, please make sure to provide @parcel/watcher as a peer dependency. Error: No prebuild or local build of @parcel/watcher found. Tried @parcel/watcher-linux-x64-glibc. 
Please ensure it is installed (don't use --no-optional when installing with npm). Otherwise it is possible we don't support your platform yet. If this is the case, please report an issue to https://github.com/parce
l-bundler/watcher.
```

Thus, I am proposing this change to the error message which improves upon the previous mentioned points:
1. It suggest that `@parcel/watcher` must be installed, not just a peer dependency.
2. It includes the error thrown during import to allow further diagnosis of the issue.

The final output is now this:

```
$ graphql-codegen --project=limes --watch
✔ Parse Configuration
✔ Generate outputs
  Failed to import @parcel/watcher due to the following error (to use watch mode, install https://www.npmjs.com/package/@parcel/watcher):
Error: No prebuild or local build of @parcel/watcher found. Tried @parcel/watcher-linux-x64-glibc. Please ensure it is installed (don't use --no-optional when installing with npm). Otherwise it is possible we don't
 support your platform yet. If this is the case, please report an issue to https://github.com/parcel-bundler/watcher.
```

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Copy-paste the updated error message in my projects `node_modules/@graphql-codegen/cli/cjs/utils/watcher.js`
- [x] Rerun the failing `graphql-codegen --watch` command

**Test Environment**:

- Docker Image: `node:18.13.0-bullseye-slim`
- `@graphql-codegen/cli`: 5.0.0

## Checklist:

- [x] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/graphql-codegen-cli/src/utils/watcher.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ export const createWatcher = (
       parcelWatcher = await import('@parcel/watcher');
     } catch (err) {
       log(
-        `Parcel watcher not found. To use this feature, please make sure to provide @parcel/watcher as a peer dependency.`
+        `Failed to import @parcel/watcher, a required dependency for watch mode, due to the following error:\n${err}`
```
Comment (n1ru4l) [inline packages/graphql-codegen-cli/src/utils/watcher.ts:69]: Maybe we should also link to the documentation or print information on how to install the dependency?

Comment (spawnia) [reply packages/graphql-codegen-cli/src/utils/watcher.ts]: Sure, I updated both the error message and the docs with a link to https://www.npmjs.com/package/@parcel/watcher.

**Review by saihaj — APPROVED** [review]

**Comment by saihaj** [comment]
@spawnia can you please run `yarn prettier . --write` and commit the changes?

**Comment by spawnia** [comment]
Sure. I found no mention of this in the contributor guide, perhaps something like https://github.com/marketplace/actions/prettier-action could help to smooth this process.

**Comment by spawnia** [comment]
How can the failing PR check be solved? I would love for this change to be included.

**Review by saihaj — APPROVED** [review]
sorry for the delay on this one

**Pull request merged**